pkg/controller/csi_handler_test.go

```diff
@@ -1381,31 +1396,56 @@ func TestCSIHandler(t *testing.T) {
 
 func TestCSIHandlerReconcileVA(t *testing.T) {
 	nID := map[string]string{
-		vaNodeIDAnnotation: testNodeName,
+		vaNodeIDAnnotation: testNodeID,
+	}
+	vaGroupResourceVersion := schema.GroupVersionResource{
+		Group:    storage.GroupName,
+		Version:  "v1",
+		Resource: "volumeattachments",
 	}
-
 	tests := []testCase{
 		{
 			name: "va attached actual state not attached",
 			initialObjects: []runtime.Object{
-				va(true /*attached*/, "" /*finalizer*/, nID /*annotations*/),
+				va(true /*attached*/, fin /* Finalizer*/, nID /*annotations*/),
 				pvWithFinalizer(),
+				csiNode(),
 			},
 			listerResponse: map[string][]string{
 				// Intentionally empty
 			},
+			expectedActions: []core.Action{
+				core.NewPatchSubresourceAction(vaGroupResourceVersion, metav1.NamespaceNone, testPVName+"-"+testNodeName,
+					types.MergePatchType, patch(va(true /*attached*/, "", nil),
+						va(true, "", nil)), "status"),
+			},
 			expectedCSICalls: []csiCall{
 				{"attach", testVolumeHandle, testNodeID, nil, nil, false, nil, false, nil, 0},
 			},
 		},
+		/* TBD
+		{
+			name:           "GCE PD va attached actual state not attached",
```

Comment:
In general, the patch looks good, is this the only thing missing?

Reply:
Here, I wanted to add a test with one of the supported translator types. Now planning to do that separately. I have created a github issue and added a TODO here.

Reply:
> In general, the patch looks good, is this the only thing missing?
I wanted to add a test with one of the supported translator types. Now planning to do that separately. I have created a github issue and added a TODO here.